src/main/java/games/strategy/engine/framework/GameDataUtils.java

```diff
@@ -33,6 +38,36 @@ public static GameData cloneGameData(final GameData data, final boolean copyDele
     }
   }
 
+  /**
+   * Clones a single GameData object multiple times.
+   * This is faster than calling {@link GameDataUtils#cloneGameData(GameData)} multiple times,
+   * because the original GameData object is only getting serialized once.
+   * 
+   * @param data The {@link GameData} to be cloned.
+   * @param times The targeted amount of {@link GameData} objects the returned {@link List} should contain.
+   * @return An unmodifiable {@link List} that containing clones of the passed data
+   */
+  public static List<GameData> cloneGameData(final GameData data, final int times) {
+    try {
+      final byte[] bytes = IoUtils.writeToMemory(os -> GameDataManager.saveGame(os, data, false));
+      return Collections.unmodifiableList(
+          IntStream.range(0, times).parallel()
+              .mapToObj(it -> {
+                try {
+                  return IoUtils.readFromMemory(bytes, GameDataManager::loadGame);
+                } catch (IOException e) {
+                  ClientLogger.logQuietly("Failed to clone game data", e);
+                  return null;
+                }
+              })
+              .filter(Objects::nonNull)
+              .collect(Collectors.toList()));
+    } catch (IOException e) {
+      ClientLogger.logQuietly("Failed to clone game data", e);
+      return Collections.emptyList();
```

Comment:
I checked out this branch and forced this error condition:

```
  public static List<GameData> cloneGameData(final GameData data, final int times) {
//    try {
          ClientLogger.logQuietly("Failed to clone game data", new IllegalStateException("simulated"));

    return Collections.emptyList();
//      final byte[] bytes = IoUtils.writeToMemory(os -> GameDataManager.saveGame(os, data, false));
//      final List<GameData> values = Collections.unmodifiableList(
//          IntStream.range(0, times)
//              .parallel()
      :
       :
   }
```

This method is called when a calc is loaded, I loaded a calc window and confirmed I saw this error message:
```
Failed to clone game data
java.lang.IllegalStateException: simulated
	at games.strategy.engine.framework.GameDataUtils.cloneGameData(GameDataUtils.java:49)
	at games.strategy.triplea.oddsCalculator.ta.OddsCalculator.setGameData(OddsCalculator.java:96)
	at games.strategy.triplea.oddsCalculator.ta.OddsCalculator.<init>(OddsCalculator.java:83)
```

The calc was active, I added a couple Russian units for attack against a german stack, and clicked calculate, I then see a never ending calculator dialog:
![infinite](https://user-images.githubusercontent.com/12397753/36077139-a32c36c2-0f1b-11e8-938f-21fe68430025.png)

Even worse, the 'cancel' button does nothing, the game is frozen in this state.

-----------------

It seems we need to enforce stronger error handling, the downstream error of game crashing due to empty result coming back is quite bad. It is perhaps simple enough to do a method post-assertion that the return set is non-empty, and on IOException instead of quiet do loud logging, and abort the calc window from displaying.